legend-sdlc-generation-service/src/main/java/org/finos/legend/sdlc/generation/service/ServiceExecutionGenerator.java

```diff
@@ -173,6 +174,28 @@ private void writeJavaClass(GeneratedJavaCode generatedJavaClass) throws IOExcep
         }
     }
 
+    private void writeEnumJavaClass(GeneratedJavaCode generatedJavaClass) throws IOException
+    {
+        Path javaClassPath = this.javaSourceOutputDirectory.resolve(getJavaSourceFileRelativePath(generatedJavaClass.getClassName()));
+        byte[] content = generatedJavaClass.getText().getBytes(StandardCharsets.UTF_8);
+        if (Files.exists(javaClassPath))
+        {
+            byte[] foundContent = Files.readAllBytes(javaClassPath);
+            if (!Arrays.equals(content, foundContent))
+            {
+                throw new IOException("Duplicate file path found with different content for enum: " + javaClassPath);
```

Comment:
I don't think IOException is the right exception type here. You could throw a FileAlreadyExistsException or just a RuntimeException.